app/controllers/register_controller.rb

```diff
@@ -1,75 +1,108 @@
 class RegisterController < ApplicationController
 
   include ApplicationHelper
-  helper_method :get_all_attribute_names_prettified, :prepare_register_name
+  helper_method :prepare_register_name, :get_description_for_register_field
 
-  # def index(register)
-  #   @countries = OpenRegister.register(register, :beta)._all_records
-  # end
+  @register_phase = Settings.register_phase
+
+  before_action :confirm, only: [:create]
 
   def index
-    @register_name = prepare_register_name(params[:register])
-    @register = OpenRegister.register(params[:register].downcase, :beta)._all_records.sort_by{|register| register.name}
+    @register = get_register(params[:register])._all_records
+    @register[0].try(:name) ? @register = @register.sort_by(&:name) : @register = @register.sort_by(&:key)
   end
 
-  def show
+  def new
+    @register = get_register(params[:register])
+    @form = JSON.parse(params.to_json)
+    render 'form'
```

Comment:
Its because there would be a load of duplication between new and edit so there is no point, just use the form. Like if you created an edit and a new view then you're just adding extra code and complexity to meet the standard rails pattern. I don't see the benefit. Following patterns for patterns sake is a bad thing. Like you're adding an injector into an injector just to meet a pattern, so you're adding render overhead to match a pattern that was designed to be used in a way that we are not using it. I think its a bad idea but happy to have the conversation if you have data to suggest why. Like the patterns that rails uses is the defaults but they include stuff to override that for a reason.